rainier-core/src/main/scala/com/stripe/rainier/core/Generator.scala

```diff
@@ -21,6 +21,12 @@ trait Generator[T] { self =>
     def get(implicit r: RNG, n: Numeric[Real]): U = fn(self.get).get
   }
 
+  def zip[U](other: Generator[U]): Generator[(T, U)] =
+    for {
```

Comment:
This definition works but it loses track of `other.requirements`. It would be better to keep them.